src/dune_rules/odoc.ml

```diff
@@ -257,51 +284,28 @@ let odoc_include_flags ctx pkg requires =
         (List.concat_map (Path.Set.to_list paths) ~f:(fun dir ->
              [ Command.Args.A "-I"; Path dir ])))
 
-let link_odoc_files sctx (odoc_file : odoc_artifact) ~pkg ~requires =
+let link_odoc_file sctx (odoc_file : odoc_artifact) ~pkg ~requires =
   let ctx = Super_context.context sctx in
   let deps = Dep.deps ctx pkg requires in
   let path =
     match pkg with
     | Some p -> Path.build (Paths.odocs ctx (Pkg p))
     | None -> Path.build (Paths.root ctx)
   in
-  let to_remove, dune_keep =
-    match odoc_file.source with
-    | Mld -> (odoc_file.odoc_input, [])
-    | Module ->
-      let dune_keep =
-        Build.create_file (odoc_file.odoc_input ++ Config.dune_keep_fname)
-      in
-      (odoc_file.odoc_input, [ dune_keep ])
-  in
   let open Build.With_targets.O in
   add_rule sctx
     ( Build.with_no_targets deps
-    >>> Build.progn
-          ( Build.with_no_targets
-              (Build.return
-                 (* Note that we declare no targets apart from [dune_keep]. This
-                    means Dune doesn't know how to build specific documentation
-                    files and that we can't run this rule in a sandbox. To
-                    properly declare targets we would need to support some form
-                    of "dynamic targets" or "target directories". *)
-                 (Action.Progn
-                    [ Action.Remove_tree to_remove
-                    ; Action.Mkdir (Path.build odoc_file.odoc_input)
-                    ]))
-          :: Command.run ~dir:path (odoc sctx)
-               [ A "link"
-               ; odoc_base_flags sctx odoc_file.odoc_input
-               ; odoc_include_flags ctx pkg requires
-                 (*TODO: am not sure if this option is appropriate here. Maybe
-                   we just need to update the flag decription in the CLI tool
-                   (odoc)*)
-               ; A "-o"
-               ; Path path
-               ; Dep (Path.build odoc_file.odoc_input)
-               ; Target odoc_file.odoc_input
-               ]
-          :: dune_keep ) )
+    >>> Command.run ~dir:path (odoc sctx)
+          [ A "link"
+          ; odoc_base_flags sctx odoc_file.odoc_input
+          ; odoc_include_flags ctx pkg requires
+            (*TODO: am not sure if this option is appropriate here. Maybe we
```

Comment:
To be discussed

Reply:
Yes, we want both base_flags and include_flags. The `-o` should be the odocl file rather than the odoc file. Other than that this is looking good!